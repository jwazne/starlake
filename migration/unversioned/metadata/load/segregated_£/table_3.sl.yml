tables:
  - name: "table_3"
    pattern: "^\\Qtable_3\\E\\d{14}(?<mode>T|D|A|)\\..*"
    attributes:
      - name: "attr1"
        type: "string"
        comment: "attr1"
      - name: "_partition"
        type: "date"
        script: "CURRENT_DATE()"
    metadata:
      sink:
        timestamp: "_partition"
        dynamicPartitionOverwrite: true
      writeStrategy:
        types:
          OVERWRITE: 'group("mode") == "T"'
          OVERWRITE_BY_PARTITION: 'group("mode") == "D" || group("mode") == ""'
          UPSERT_BY_KEY: 'group("mode") == "A"'
        type: "OVERWRITE_BY_PARTITION"
  - name: "table_4"
    pattern: "^\\Qtable_5\\E\\d{14}(?<mode>T|D|A|)\\..*"
    attributes:
      - name: "attr1"
        type: "string"
        comment: "attr1"
      - name: "_partition"
        type: "date"
        script: "CURRENT_DATE()"
    metadata:
      sink:
        timestamp: "_partition"
    merge:
      key:
        - "key1"
        - "key2"