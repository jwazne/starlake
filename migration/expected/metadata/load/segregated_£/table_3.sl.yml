---
version: 1
table:
  name: "table_3"
  pattern: "^\\Qtable_3\\E\\d{14}(?<mode>T|D|A|)\\..*"
  attributes:
    - name: "attr1"
      type: "string"
      comment: "attr1"
    - name: "_partition"
      type: "date"
      script: "CURRENT_DATE()"
  metadata:
    sink:
      partition:
        - "_partition"
    writeStrategy:
      types:
        OVERWRITE: 'group("mode") == "T"'
        OVERWRITE_BY_PARTITION: 'group("mode") == "D" || group("mode") == ""'
        DELETE_THEN_INSERT: 'group("mode") == "A"'
      type: "OVERWRITE_BY_PARTITION"